### PR TITLE
Added examples in weakly_connected.py

### PR DESCRIPTION
**Pull request by NikitaSharma1**
Added examples to the following functions:
- number_weakly_connected_components()
- is_weakly_connected()

### CODE REVIEW

File: networkx/algorithms/components/weakly_connected.py [hunk 1]
```diff
@@ -83,6 +83,14 @@ def number_weakly_connected_components(G):
     NetworkXNotImplemented
         If G is undirected.
 
+    Examples
+    --------
+    >>> import networkx as nx
```
Comment (rossbar) [inline networkx/algorithms/components/weakly_connected.py:88]: This is done automatically for all doctests so doesn't need to be specified explicitly.

```suggestion
```

File: networkx/algorithms/components/weakly_connected.py [hunk 2]
```diff
@@ -83,6 +83,14 @@ def number_weakly_connected_components(G):
...
+    >>> G = nx.DiGraph()
+    >>> G.add_edges_from([(0, 1), (2, 1), (3, 4)])
```
Comment (rossbar) [inline networkx/algorithms/components/weakly_connected.py:90]: IMO it's better to keep the example as simple as possible, so we can get rid of the explicit `add_edges_from` call:

```suggestion
    >>> G = nx.DiGraph([(0, 1), (2, 1), (3, 4)])
```

File: networkx/algorithms/components/weakly_connected.py [hunk 3]
```diff
@@ -123,6 +131,18 @@ def is_weakly_connected(G):
     NetworkXNotImplemented
         If G is undirected.
 
+    Examples
+    --------
+    >>> import networkx as nx
+    >>> G = nx.DiGraph()
+    >>> G.add_nodes_from([0, 1, 2, 3])
+    >>> G.add_edges_from([(0, 1), (2, 1)])
```
Comment (rossbar) [inline networkx/algorithms/components/weakly_connected.py:139]: Similar comment as above, though in this case accounting for the addition of the unconnected node:

```suggestion
    >>> G = nx.DiGraph([(0, 1), (2, 1)])
    >>> G.add_node(3)
```

Comment (NikitaSharma1) [reply networkx/algorithms/components/weakly_connected.py]: I have made the changes.

**Review by dschult — APPROVED** [review]
This LGTM!
Thanks

**Pull request merged**